src/Native/Common/pal_config.h.in

```diff
@@ -36,6 +36,9 @@
 #cmakedefine01 HAVE_TCP_VAR_H
 #cmakedefine01 HAVE_RT_MSGHDR
 #cmakedefine01 HAVE_LINUX_RTNETLINK_H
+#cmakedefine01 HAVE_INOTIFY_INIT
```

Comment:
We shouldn't check explicitly for linux...rather, we should just look for the headers and if the HAVE_INOTIFY value isn't defined, in the pal_io.cpp code below where you check #if HAVE_INOTIFY, you can have an else that is simply #error

We should try to avoid looking for specific operating systems as much as possible and do feature-based configuration wherever possible :)